kani-driver/src/call_cbmc.rs

```diff
@@ -102,6 +102,10 @@ impl KaniSession {
 
         args.push("--slice-formula".into());
 
+        if self.args.add_exe_trace_to_src {
```

Comment:
That's a really good catch. I was surprised when I saw this, because I didn't know why my test cases where passing since they seemed to be getting det vals when they only used `--gen-exe-trace`. I did a bit of digging, and this is what I found: `cbmc` appears to automatically add the trace when it runs with `--json-ui`. Since `--json-ui` is automatically enabled for the default output format, the trace is included with the default Kani configs. This might be relevant to #1288 @jaisnan.